src/Orleans.EventSourcing/Hosting/StateStorageSiloBuilderExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Orleans.LogConsistency;
+using Orleans.Providers;
+using Orleans.Runtime;
+using Orleans.EventSourcing.StateStorage;
+
+namespace Orleans.Hosting
+{
+    public static class StateStorageSiloBuilderExtensions
+    {
+        /// <summary>
+        /// Adds a state storage log consistency provider as default consistency provider"/>
+        /// </summary>
+        public static ISiloHostBuilder AddStateStorageBasedLogConsistencyProviderAsDefault(this ISiloHostBuilder builder)
+        {
+            return builder.AddStateStorageBasedLogConsistencyProvider(ProviderConstants.DEFAULT_STORAGE_PROVIDER_NAME);
+        }
+
+        /// <summary>
+        /// Adds a state storage log consistency provider"/>
+        /// </summary>
+        public static ISiloHostBuilder AddStateStorageBasedLogConsistencyProvider(this ISiloHostBuilder builder, string name = "StateStorage")
+        {
+            return builder.ConfigureServices(services => services.AddStateStorageBasedLogConsistencyProvider(name));
+        }
+
+        internal static IServiceCollection AddStateStorageBasedLogConsistencyProvider(this IServiceCollection services, string name)
```

Comment:
this method does the same thing as `AddLogStorageBasedLogConsistencyProvider`. Why is this ? am I missing anything ?

Reply:
It adds Orleans.EventSourcing.StateStorage.LogConsistencyProvider while AddLogStorageBasedLogConsistencyProvider adds Orleans.EventSourcing.LogStorage.LogConsistencyProvider